Neos.ContentGraph.DoctrineDbalAdapter/src/ContentGraphFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter;
+
+use Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository\ContentGraph;
+use Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository\NodeFactory;
+use Neos\ContentRepository\Core\ContentGraphFactoryInterface;
+use Neos\ContentRepository\Core\Infrastructure\DbalClientInterface;
+use Neos\ContentRepository\Core\NodeType\NodeTypeManager;
+use Neos\ContentRepository\Core\Projection\Workspace\Workspace;
+use Neos\ContentRepository\Core\Projection\Workspace\WorkspaceStatus;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamDoesNotExistYet;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceDescription;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceTitle;
+
+/**
+ * @internal only used inside the
+ * @see ContentGraphFinder
+ */
+readonly final class ContentGraphFactory implements ContentGraphFactoryInterface
+{
+    public function __construct(
+        private DbalClientInterface $client,
+        private NodeFactory $nodeFactory,
+        private ContentRepositoryId $contentRepositoryId,
+        private NodeTypeManager $nodeTypeManager,
+        private string $tableNamePrefix
+    ) {
+    }
+
+    public function buildForWorkspace(WorkspaceName $workspaceName): ContentGraph
+    {
+        // FIXME: Should be part of this projection, this is forbidden
```

Comment:
> note the big fat FIXME in both adapters factory implementations as those now require usage of a table that connects workspce to csId, and given that that is not there yet I hacked it, obviously thís needs to change

Reply:
I just realized this morning that I _could_ possibly move this resolving into the ContentGraphFinder which could have legitimate access to a workspace finder, just not sure I can easily resolve this dependency...

Reply:
> Either way we want this in the CG projection (adapters) anyways.

yes... thesis is everything that is used for validating commands (soft constraints) must be now provided by the one cg projection so everything is more consistent and less flacky.

Reply:
I think we can merge this first with these changes and followup by embedding the projection